### PR TITLE
added conversion functions with test cases along with vendor and tool…

### PR DESCRIPTION
**Pull request by shantanu7p**
…s folder

### CODE REVIEW

**Comment by kalehrishi** [comment]
`Convertor` spelling is wrong. It should be `Converter`

File: app/LengthConvertor.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	/*
+	 *This function Converts the Input Length from Meter To Kilometer
+	 */
+	public function convertMeterToKilometer($length) {
+		if (is_string($length)) {
+			throw new \InvalidArgumentException("Please Enter valid Length");
+		} elseif (is_null($length)) {
+			throw new \InvalidArgumentException("Please Enter valid Length");
+		}
+		$KiloMeter = $length/1000;
```
Comment (kalehrishi) [inline app/LengthConvertor.php:13]: follow camel case. `$KiloMeter` is variable it should be named `$kiloMeter` or `$kilometer`

File: app/TemperatureConvertor.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?php
+namespace ConvertorApp;
+class TemperatureConvertor {
+	/*
+	 * This function Converts the Input temprature from fahrenhit to celsius
+	*/
+	public function convertFahrenhitToCelsius($temprature) {
```
Comment (kalehrishi) [inline app/TemperatureConvertor.php:7]: Wrong spelling for `Fahrenhit`. It should be  `Fahrenheit`.

**Comment by shantanu7p** [comment]
is it possible to rename the repository?

File: app/TemperatureConvertor.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		if (is_string($temprature)) {
```
Comment (kalehrishi) [inline app/TemperatureConvertor.php:8]: Wrong spelling `temperature`

File: app/LengthConvertor.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	public function convertCentimeterToMeter($length) {
+		if (is_string($length)) {
+			throw new \InvalidArgumentException("Please Enter valid Length");
+		} elseif (is_null($length)) {
+			throw new \InvalidArgumentException("Please Enter valid Length");
+		}
+		$Meter = $length/100;
+		return $Meter;
+	}
+}
```
Comment (kalehrishi) [inline app/LengthConvertor.php:52]: End file with new line.

File: app/WeightConvertor.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<?php
+namespace ConvertorApp;
+class WeightConvertor {
+	/*
+	 * This function Converts the Input Weight from Gram To Kilogram
+	 */
+	public function convertGramToKilogram($weight) {
+		if (is_string($weight)) {
```
Comment (kalehrishi) [inline app/WeightConvertor.php:8]: All your validations are same. You can create a base class and add validate function to that class and inherit that class in each `Convertor` class. Will save lot of code.

File: tests/LengthConvertorTest.php [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+   $convert = new LengthConvertor();
+	 $convert->convertMeterToKilometer("abc");
+	}
+  /**
+	 *@test
+	 */
+  public function convertMeterToKilometer_function_can_validate_negative_integer_argument() {
+	 $convert = new LengthConvertor();
+	 $result  = $convert->convertMeterToKilometer(-1);
+	 $answer = $result-(-0.001);
```
Comment (kalehrishi) [inline tests/LengthConvertorTest.php:27]: Where is assertion?

File: tests/LengthConvertorTest.php [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+namespace ConvertorApp;
+class LengthConvertorTest extends \PHPUnit_Framework_TestCase {
+  /**
+	 *@test
+	 */
+	public function convertMeterToKilometer_function_can_validate_integer_value() {
+		$convert = new LengthConvertor();
+		$result  = $convert->convertMeterToKilometer(1);
+		$answer = $result-(0.001);
+		$this->assertLessThan(0.0001, $answer);
```
Comment (kalehrishi) [inline tests/LengthConvertorTest.php:11]: Why less than and not equal?

File: tests/LengthConvertorTest.php [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+	}
+	/**
+	 *@test
+	 * @expectedException InvalidArgumentException
+	 */
```
Comment (kalehrishi) [inline tests/LengthConvertorTest.php:16]: You should also test `@expectedExceptionMessage`

File: tests/LengthConvertorTest.php [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+  public function convertKilometerToMeter_function_should_throw_InvalidArgumentException_for_argument_which_Contain_string() {
+		$convert = new LengthConvertor();
+		$convert->convertKilometerToMeter("abc");
+		}
+   /**
+	 *@test
+	 */
+  public function convertKilometerToMeter_function_can_validate_negative_integer_argument() {
```
Comment (kalehrishi) [inline tests/LengthConvertorTest.php:56]: Whats negative distance?

File: tests/LengthConvertorTest.php (hunk 6 above)
Comment (shantanu7p) [inline tests/LengthConvertorTest.php:27]: for string checking test cases function throws exception , then why we need assertion for that?

File: tests/temperatureConvertorTest.php [hunk 10]
```diff
@@ -0,0 +1,138 @@
+<?php
+namespace ConvertorApp;
+class temperatureConvertorTest extends \PHPUnit_Framework_TestCase {
```
Comment (kalehrishi) [inline tests/temperatureConvertorTest.php:3]: Class name should start with Capital letter.

File: tests/LengthConvertorTest.php (hunk 7 above)
Comment (shantanu7p) [inline tests/LengthConvertorTest.php:11]: sir for float validation we need to check that condition using assertLessThan .

**Pull request closed without merging**